aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,639 @@
+    use aptos_framework::timestamp;
+    use aptos_std::bucket_table;
+    use aptos_framework::coin;
+    use aptos_framework::aptos_coin::AptosCoin;
+    use aptos_framework::account;
+    use std::bcs;
+    #[test_only]
+    use aptos_token::token::create_token_script;
+    use dao_platform::nft_dao_events::emit_create_dao_event;
+
+    /// Only DAO adminstrators can initialize the DAO
+    const EONLY_ADMIN_CAN_INITIALIZE_DAO: u64 = 1;
+
+    /// Only maximal = 1 and maximal immutable token can be used for DAO
+    const ENOT_NFT_TOKEN: u64 = 2;
+
+    /// This NFT has already enrolled in DAO
+    const ETOKEN_ALREADY_ENROLLED_IN_DAO: u64 = 3;
+
+    /// This account doesn't have enough voting weights
+    const EVOTING_WEIGHTS_NOT_ENOUGH: u64 = 4;
+
+    /// This account doesn't own this DAO's voting token
+    const ENOT_OWN_THE_VOTING_DAO_TOKEN: u64 = 5;
+
+    /// This function is supported in proposal
+    const ENOT_SUPPROTED_FUNCTION: u64 = 6;
+
+    /// Can only propose a start time in future
+    const EPROPOSAL_START_TIME_SHOULD_IN_FUTURE: u64 = 7;
+
+    /// Cannot only use membership token of the DAO
+    const ENOT_A_MEMBER_TOKEN: u64 = 8;
+
+    /// String length exceeds limits
+    const ESTRING_TOO_LONG: u64 = 9;
+
+    /// Proposal ended
+    const EPROPOSAL_ENDED: u64 = 10;
+
+    /// Propsoal already resolved
+    const EPROPOSAL_RESOLVED: u64 = 11;
+
+    /// Denominator should be bigger than 0
+    const EDEMONINATOR_CANNOT_BE_ZERO: u64 = 12;
+
+    /// Denominator should be bigger or equal to numerator
+    const ENUMERATOR_BIGGER_THAN_DENOMINATOR: u64 = 13;
+
+    /// NOT ENOUGH PEOPLE VOTED
+    const ENOT_ENOUGH_TOKEN_VOTED: u64 = 14;
+
+    /// TOKEN VOTED FOR PROPOSAL
+    const ETOKEN_ALREADY_VOTED: u64 = 15;
+
+    /// Resource Account Doesn't exist
+    const ERESOURCE_ACCT_NOT_EXIST: u64 = 16;
+
+    /// Proposal resource not created
+    const EPRPOSALS_NOT_EXIST_AT_ADDRESS: u64 = 17;
+
+    /// Proposal id doesn't exist
+    const EPRPOSAL_ID_NOT_EXIST: u64 = 18;
+
+
+    /// Constants
+    const PROPOSAL_UNRESOLVED: u8 = 0;
+    const PROPOSAL_RESOLVED_PASSED: u8 = 1;
+    const PROPOSAL_RESOLVED_NOT_PASSED: u8 = 2;
+
+    const DEFALT_VOTING_WEIGHT: u64 = 1;
+
+    struct DAO has key {
+        /// Name of the DAO
+        name: String,
+        /// The threshold that the proposal can resolve. For example, 50% means at least 50% of NFT holder should vote to resolve the proposal
+        resolve_threshold: ResolveThreshold,
+        /// The NFT Collection that is used to govern the DAO
+        governance_token: GovernanceToken, // This is the governance token or NFT used for voting
+        /// The address of the creator account this DAO owns. THis account is a resource account that this DAO is an adminstrator of the controlled account
+        controlled_account: address, // This is the nft creator account that this DAO governs, this accout should offer it signer cap to DAO account
```

Comment:
Is the intention here to use this address to get the required signercap in RegisteredAccounts for control over the creation later?

Reply:
no, the controlled_account was for locating a creator resource account that grant admin to this DAO account. But I think we don't want this in v0. deleted it.

Reply:
It could be in v0 though? We just need a flow for people to deposit the SignerCapability for the collection creator account right? That seems relatively simple to me but we can double check with the community (e.g. Aries) to see if it's easy to support in the UI.

Reply:
It is infeasible now. The SignerCapability of the creator resource account is stored at the resource account itself.

Reply:
Why is it infeasible? The contract that holds the creator's SignerCap just needs to support extracting it right?

Reply:
If you extract the signer_cap out of creator resource account, how to handle mint request?